src/sgame/sg_bot_ai.cpp

```diff
@@ -869,37 +868,41 @@ AINodeStatus_t BotActionFight( gentity_t *self, AIGenericNode_t *node )
 	// direct navmesh path and we are not at at weapon range (if human)
 
 	BotAimAtEnemy( self );
+	glm::vec3 dummy;
+	//TODO when failing to handle obstacle, a different target should be taken
+	//  for now, at least avoid wasting stamina
+	canSprint = !BotAvoidObstacles( self, dummy, false );
 	BotMoveInDir( self, MOVE_FORWARD );
 
-	if ( inAttackRange || self->client->ps.weapon == WP_PAIN_SAW )
+	if ( equipedInRange )
 	{
 		BotFireWeaponAI( self );
 	}
 
 	if ( myTeam == TEAM_ALIENS )
 	{
-		BotClassMovement( self, inAttackRange );
+		BotClassMovement( self, primaryWP );
```

Comment:
The 2nd argument is supposed to be a bool

Reply:
indeed, way a copy/pasta error